### ISSUE TITLE
Unescaped control characters on regexp constructors

### ISSUE THREAD

**Issue by pepagos** [babel/babili#268]
The plugin babel-plugin-transform-regexp-constructors doesn't correctly manages regular expressions with control characters. In fact, \n is emitted verbatim generating illegal regexps. This only happens with new RegExp("\n") and not with /\n/.

The following mocha+should test captures the problem:
````javascript
describe("babili", function() {
  describe("has a bug", function() {
   it("with regexps that have control characters", function() {
      var code = "var x = /hello\\nbye/";
      var min = babel.transform(code, {presets:["babili"], comments:false})
        .code;
      min.should.match(/hello\\nbye/);
      eval(min);
      var code2 = 'var x = new RegExp("hello\\nbye")';
      var min2 = babel.transform(
        code2, {presets:["babili"], plugins: ["remove-comments"]}).code;
      min2.should.match(/hello\nbye/);
      (() => eval(min2)).should
        .throw(/Invalid regular expression: missing \//);
    });
  });
````

**Comment by kangax** [comment]
This is already fixed in master

**Comment by pepagos** [comment]
Thus, it should be solved in 0.0.9, shouldn't it?
But with that version I see the same problem:

$ node
> var babel = require('babel-core')
undefined
> babel.transform("new RegExp(\"hello\\nbye\")", {presets:["babili"]}).code
'/hello\nbye/;'

It should return '/hello\\nbye/'

**Comment by kangax** [comment]
Are you sure you're on latest? I just checked and I'm seeing "\\n"

**Comment by pepagos** [comment]
My project had a dependency on 

    "dependencies": {
      "babel-core": "^6.18.2",
      "babili": "=0.0.9"
    }

Those are the only dependencies. I remove node_modules and execute npm install and what I get is babili@**0.0.9** and babel-preset-babili@**0.0.8**. I don't know why the preset is in version 0.0.8.
If change dependencies to 

    "dependencies": {
      "babel-core": "^6.18.2",
      "babel-preset-babili": "=0.0.9"
    }

now I get babel-preset-babili@0.0.9 and everything works as expected.

So, you are right: it is corrected in 0.0.9, but for some reason babili 0.0.9 installs babel-preset-babili 0.0.8.

Sorry for the inconveniences. Thank you for your patience.
